DigitalLearningSolutions.Data/Services/ActivityService.cs

```diff
@@ -126,6 +132,28 @@ public byte[] GetActivityDataFileForCentre(int centreId, ActivityFilterData filt
             return stream.ToArray();
         }
 
+        public (DateTime startDate, DateTime? endDate)? GetValidatedUsageStatsDateRange(
+            string startDateString,
+            string endDateString,
+            int centreId
+        )
+        {
+            var parsedStartDate = DateTime.Parse(startDateString);
+            var start = GetStartOfActivityForCentre(centreId);
+            if (parsedStartDate < GetStartOfActivityForCentre(centreId))
+            {
+                return null;
+            }
+
+            if (DateTime.TryParse(endDateString, out var parsedEndDate) &&
+                (parsedEndDate < parsedStartDate || parsedEndDate > DateTime.Now))

```

Comment:
This is quite long and hard to read, maybe we can pull `var isEndDateSet `into a separate var?